runtime/parser/parser.go

```diff
@@ -231,6 +231,13 @@ func (p *parser) next() {
 	}
 }
 
+// utility for the common pattern of advancing past the current token to
+// the next semantic token
```

Comment:
By convention, Go documentation comments should start with the name of the declaration:

```suggestion
// nextSemanticToken advances past the current token to the next semantic token.
// It skips whitespace, including newlines, and comments
```